File: snippets/java/Snippet-003.java
import java.io.FileNotFoundException;
import java.io.PrintWriter;

class Snippet {

    public static void main(String[] args) throws FileNotFoundException {
        PrintWriter écrivain = new PrintWriter("customer-01");
        écrire(écrivain, "firstname", "Olivier");
        écrire(écrivain, "lastname", "PEREZ");
        écrivain.close();
    }

    private static void écrire(PrintWriter output, String key, Object value) {
        output.write(key + "(" + value.getClass() + ")=" + value);
    }
}

File: kotlin/snippets/Snippet-003.kt
package fr.o80.kata

import java.io.PrintWriter

fun main() {
    val printWriter = PrintWriter("customer-01")
    put(printWriter, "firstname", "Olivier")
    put(printWriter, "lastname", "PEREZ")
    printWriter.close()
}

fun put(output: PrintWriter, key: String, value: Any) {
    output.write("$key(${value.javaClass})=$value")
}
